entity-service-impl/src/main/java/org/hypertrace/entity/query/service/converter/ValueHelper.java

```diff
@@ -346,7 +346,8 @@ private static Map<String, ValueType> getStringValueToPrimitiveTypeMap() {
 
   private JSONDocument convertToDocument(final Value value) throws ConversionException {
     try {
-      return new JSONDocument(PRINTER.print(LiteralConstant.newBuilder().setValue(value)));
+      return new JSONDocument(
+          PRINTER.print(LiteralConstant.newBuilder().setValue(value.toBuilder().clearValueType())));
```

Comment:
Thanks, Aaron. Added in the description

Reply:
Thanks - So that explains why it wasn't happening for strings. I'm guessing that ValueHelper, even though it doesn't say it, is only used for bulk writes not other updates (else we would have hit this before)? 

Also, this would help for new data - but wouldn't we need to fix the read path for data that has already been written?

Reply:
>  I'm guessing that ValueHelper, even though it doesn't say it, is only used for bulk writes not other updates (else we would have hit this before

Yes.

> but wouldn't we need to fix the read path for data that has already been written?

Problematic documents have been corrected in the DB.